shamil_latypov/logger.cpp

```diff
@@ -0,0 +1,26 @@
+#include "logger.hpp"
+#include <assert.h>
+#include <cassert>
+#include <chrono>
+#include <iomanip>
+#include <iostream>
+#include <sstream>
+
+namespace uni_cource_cpp {
+
+void Logger::set_file_output_name(const std::string& file_path) {
+  file_out_ = std::ofstream(file_path);
+  assert(file_out_.is_open() && "File didnt open. Maybe Temp dir soesnt exist");
```

Comment:
Это не баг в коде - это уже потенциально взаимодействие с операционной системой. В таких ситуация нужно использовать `exception` а не `assert`.
```suggestion
  throw std::runtime_error("Failed to create file stream");
```